`bytemuck` dependency `bytemuck_derive` min version `1.4.0` has build errors with generic structs

I encountered a build failure in a dependency (`skrifa v0.22.3`) that uses `bytemuck`, and the problem turned out to be that our workspace `Cargo.lock` had an older version of `bytemuck_derive`. 

We don't depend on either of these crates directly (only through other dependencies), we have never attempted to pin the version of either one, so when a dependency came along and required a more recent version of `bytemuck`, it failed to update `bytemuck_derive` and the build started to fail in a confusing way.

Here's a way to reproduce the problem:
```toml
[package]
name = "test_bytemuck"
version = "0.1.0"
edition = "2021"

[dependencies]
bytemuck = { version = "1.21.0", features = ["derive"] }
```

lib.rs:
```rust
#[derive(Clone, Copy, bytemuck::AnyBitPattern)]
pub struct Nullable<T>(T);
```

I can then reproduce the build failure by placing an old version of `bytemuck_derive` in my `Cargo.lock`.
```plaintext
cargo update -p bytemuck_derive --precise 1.4.0
```

This results in the dependency tree:
```plaintext
test_bytemuck v0.1.0
    └── bytemuck_derive v1.4.0 (proc-macro)
        ├── proc-macro2 v1.0.93
        │   └── unicode-ident v1.0.16
        ├── quote v1.0.38
        │   └── proc-macro2 v1.0.93 (*)
        └── syn v1.0.109
            ├── proc-macro2 v1.0.93 (*)
            ├── quote v1.0.38 (*)
            └── unicode-ident v1.0.16
```

And the build fails in a confusing way:
```plaintext
$ cargo check
    Checking test_bytemuck v0.1.0
error[E0107]: missing generics for struct `Nullable`
 --> src/lib.rs:4:12
  |
4 | pub struct Nullable<T>(T);
  |            ^^^^^^^^ expected 1 generic argument
  |
note: struct defined here, with 1 generic parameter: `T`
 --> src/lib.rs:4:12
  |
4 | pub struct Nullable<T>(T);
  |            ^^^^^^^^ -
help: add missing generic argument
  |
4 | pub struct Nullable<T><T>(T);
  |                    +++

error[E0277]: the trait bound `Nullable<T>: Zeroable` is not satisfied
  --> src/lib.rs:4:12
   |
4  | pub struct Nullable<T>(T);
   |            ^^^^^^^^^^^ the trait `Zeroable` is not implemented for `Nullable<T>`
   |
note: required by a bound in `AnyBitPattern`
  --> /home/eric/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bytemuck-1.21.0/src/anybitpattern.rs:52:3
   |
51 | pub unsafe trait AnyBitPattern:
   |                  ------------- required by a bound in this trait
52 |   Zeroable + Sized + Copy + 'static
   |   ^^^^^^^^ required by this bound in `AnyBitPattern`

Some errors have detailed explanations: E0107, E0277.
For more information about an error, try `rustc --explain E0107`.
error: could not compile `test_bytemuck` (lib) due to 2 previous errors
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -93,7 +93,7 @@ nightly_float = []
 nightly_docs = []
 
 [dependencies]
-bytemuck_derive = { version = "1.4", path = "derive", optional = true }
+bytemuck_derive = { version = "1.4.1", path = "derive", optional = true }
 
 [lints.rust]
 unexpected_cfgs = { level = "deny", check-cfg = ['cfg(target_arch, values("spirv"))'] }

